curve_apps/driver.py

```diff
@@ -87,17 +86,15 @@ def params(self, val: BaseData):
             raise TypeError("Parameters must be of type Parameters.")
         self._params = val
 
-    def add_ui_json(self, entity: ObjectBase | ContainerGroup):
+    def add_ui_json(self, entity: ObjectBase | ContainerGroup) -> None:
         """
         Add ui.json file to entity.
 
         :param entity: Object to add ui.json file to.
         """
-        param_dict = self.params.flatten()
-        self.params.input_file.update_ui_values(param_dict)
 
         with tempfile.TemporaryDirectory() as temp_dir:
-            file_path = self.params.input_file.write_ui_json(
-                path=temp_dir, name=self.params.name + ".ui.json"
-            )
-            entity.add_file(str(file_path))
+            filepath = Path(temp_dir) / f"{self.params.name}.ui.json"
+            self.params.write_ui_json(filepath)
+
+            entity.add_file(str(filepath))
```

Comment:
Should try to streamline this on https://mirageoscience.atlassian.net/browse/GEOPY-1536

Reply:
yes, sounds good